src/target/riscv/riscv-013.c

```diff
@@ -1272,193 +1286,224 @@ static bool has_sufficient_progbuf(struct target *target, unsigned size)
 	return info->progbufsize + r->impebreak >= size;
 }
 
-/**
- * Immediately write the new value to the requested register. This mechanism
- * bypasses any caches.
+/** This function recives a program that stores a register at a memory address
+ * stored at S0, allocates the memory needed, executes it and reads the value
+ * of the register.
  */
```

Comment:
Thanks for the comment. I'd suggest slightly different wording:

```suggestion
/** 
 * This function allocates 64 bits of scratch memory, writes its address to s0 and executes program buffer.
 * Then it reads the value of the stratch memory.
 * The scratch memory is deallocated at the end.
 *
 * Use case: Reading of D-extension registers (64-bit) on RV32, which needs to be done through scratch memory.
 *
 * The caller should save s0.
 * The caller should prepare a suitable program that fills the scratch memory location.
 */
```

Reply:
IMHO, this is an excessively long comment for such a small and local function. However, my wording was really poor. I have rewritten the comment.

Reply:
Looks all right - thanks.